src/value-indexer/src/main/scala/ValueIndexer.scala

```diff
@@ -151,7 +149,10 @@ class ValueIndexerModel(val uid: String)
   private def addCategoricalColumn[T: TypeTag](dataset: Dataset[_])
                                               (implicit ct: ClassTag[T]): DataFrame = {
     val nonNullLevels = getLevels.filter(_ != null)
-    val castLevels = nonNullLevels.map(_.asInstanceOf[T])
+    val castLevels = nonNullLevels.map {
+      case l: scala.math.BigInt => l.toInt.asInstanceOf[T]
```

Comment:
why is the BigInt case needed in the map?

Reply:
BigInt does not cast to an Int, it needs to be converted before coercion.

Reply:
I don't think there would ever be a BigInt in this case though.  Was there an exception that was thrown that led you to add this case?  Can the case be removed?

Reply:
yes I had to add this to fix an error